lua/telescope/_extensions/laravel.lua

```diff
@@ -213,9 +219,128 @@ local routes = function(opts)
     :find()
 end
 
+local related = function(opts)
+  opts = opts or {}
+
+  local file_type = vim.bo.filetype
+  local lang = vim.treesitter.language.get_lang(file_type)
+  if lang ~= "php" then
+    return false
+  end
+
+  local get_model_class_name = function()
+    local query = vim.treesitter.query.parse(
+      lang,
+      [[ (namespace_definition name: (namespace_name) @namespace)
+  	(class_declaration name: (name) @class) ]]
+    )
+    local tree = vim.treesitter.get_parser():parse()[1]:root()
+    local bufNr = vim.fn.bufnr()
+    local class = ""
+    for id, node, _ in query:iter_captures(tree, bufNr, tree:start(), tree:end_()) do
+      if query.captures[id] == "class" then
+        class = class .. "\\" .. vim.treesitter.get_node_text(node, 0)
+      elseif query.captures[id] == "namespace" then
+        class = vim.treesitter.get_node_text(node, 0) .. class
+      end
+    end
+    return class
+  end
+
+  local class = get_model_class_name()
+  if class ~= "" then
+    local result, ok = application.run("artisan", { "model:show", class, "--json" }, { runner = "sync" })
+    if not ok then
+      return nil
+    end
+
+    if result.exit_code ~= 0 or string.sub(result.out[1], 1, 1) ~= "{" or string.sub(result.out[1], -1) ~= "}" then
+      return nil
```

Comment:
The idea is that we don't really need to check for the model parent class, because the artisan command does it for us, as it expects the provided class to have some methods. We can notify the user though that command errors probably due to the class but being the Model.

We can add the check, I am just not sure we really need it.